### PR TITLE
[SPARK-38727][SQL][TESTS] Test the error class: FAILED_EXECUTE_UDF

### PR DESCRIPTION
**Pull request by panbingkun**
## What changes were proposed in this pull request?
This pr aims to add one test for the error class FAILED_EXECUTE_UDF to QueryExecutionErrorsSuite.

### Why are the changes needed?
The changes improve test coverage, and document expected error messages in tests.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
By running new test:
```
build/sbt "sql/testOnly *QueryExecutionErrorsSuite*"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -388,4 +388,19 @@ class QueryExecutionErrorsSuite extends QueryTest
...
+      val hello = udf { _: String => {
+        throw new SparkException("throw spark exception for udf test")
+      }}
+      names.select(hello($"name")).collect()
+    }
+    assert(e1.getCause.isInstanceOf[SparkException])
+
+    val e2 = e1.getCause.asInstanceOf[SparkException]
+    assert(e2.getErrorClass === "FAILED_EXECUTE_UDF")
+    assert(e2.getMessage.startsWith("Failed to execute user defined function "))
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:404]: Please, compare to the entire error message. A purpose is to document errors in tests.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala]: ok, i will use regex match it.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 2]
```diff
@@ -389,15 +389,19 @@ class QueryExecutionErrorsSuite extends QueryTest
     }
   }
 
-  test("INVALID_FRACTION_OF_SECOND: in the function make_timestamp") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:392]: Why did you remove the test?

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 3]
```diff
@@ -389,15 +389,19 @@ class QueryExecutionErrorsSuite extends QueryTest
...
-      }
-      assert(e.getErrorClass === "INVALID_FRACTION_OF_SECOND")
-      assert(e.getSqlState === "22023")
-      assert(e.getMessage === "The fraction of sec must be zero. Valid range is [0, 60]. " +
-        "If necessary set spark.sql.ansi.enabled to false to bypass this error. ")
+  test("FAILED_EXECUTE_UDF: execute user defined function") {
+    val e1 = intercept[SparkException] {
+      val names = Seq("Jacek", "Agata", "Sweet").toDF("name")
+      val hello = udf { _: String => {
+        throw new SparkException("throw spark exception for udf test")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:396]: User's functions shouldn't throw the Spark exception. Let's make the test more realistic and throw some other Java exception.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — APPROVED** [review]
Waiting for https://github.com/panbingkun/spark/actions/runs/2189497119

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 4]
```diff
@@ -418,4 +418,20 @@ class QueryExecutionErrorsSuite extends QueryTest
...
+        word.substring(index, index + 1)
+      }}
+      words.select(luckyCharOfWord($"word", $"index")).collect()
+    }
+    assert(e1.getCause.isInstanceOf[SparkException])
+
+    val e2 = e1.getCause.asInstanceOf[SparkException]
+    assert(e2.getErrorClass === "FAILED_EXECUTE_UDF")
+    assert(e2.getMessage.matches("Failed to execute user defined function " +
+      "\\(QueryExecutionErrorsSuite\\$\\$Lambda\\$\\d+/\\d+: \\(string, int\\) => string\\)"))
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:435]: This seems to break Java 17 tests, @panbingkun and @MaxGekk . Please see the following community patch.
- https://github.com/apache/spark/pull/36270
